### DISCUSSION TITLE
Support for CIDR column type in Migrations?

### DISCUSSION THREAD

**Discussion by rahulmkhj** [laravel/framework#53776] [Ideas]
PostgreSQL supports two types of columns to store IP/Network information, inet and cidr. In laravel migrations we can add an inet column via ipAddress(). However, I don't seem to find a method to create a CIDR type column. 

May I request to add support for CIDR type of columns?

**Comment by hafezdivandari** [comment]
You may use `rawColumn()` added on v11.32.0 via #53496:

```php
Schema::create('my_table', function (Blueprint $table) {
    $table->id();
    $table->rawColumn('my_column', 'cidr');
});
```

**Comment by shaedrich** [comment]
Would you implement this as a new method on the blueprint class or as an additional argument for the existing `ipAddress()` method?

**Reply by rahulmkhj** [reply]
Personally I don't have a preference. But looking at the currently supported column types here: https://laravel.com/docs/11.x/migrations#available-column-types we have separate methods for timestamps and nullableTimestamps and so on.. A new method would be relatively more suitable :)

**Reply by shaedrich** [reply]
Wouldn't say, I like that, but you've got a point 👍🏻 

[I checked the actual code and there are some methods (e.g. `unsignedInteger()`) which internally call another method with pre-set arguments](https://github.com/laravel/framework/blob/a30619310dd739c22f561579c3de07f33d109612/src/Illuminate/Database/Schema/Blueprint.php#L958-L961). That might be another possible approach that would combine both by keeping them separate in the public API and combining them as an implementation detail 🤔
